enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,871 @@
+
+If `controlPlaneCPULimit >= DefaultControlPlaneCPUCount`, operators
+should use their default configuration for their operands. We do not
+currently have a need to allow the control plane to scale up, although
+this may change later. If we do have a future requirement to support
+that, we could remove this restriction.
+
+The logic described in this section could also be captured in a
+library function. Something like
+
+```go
+cpuRequestValue, err := GetScaledCPURequest(kubeClientConfig, myDefaultValue)
+```
+
+where the function fetches the Infrastructure resource and applies the
+algorithm to a default value provided by the caller. That would make
+the code change to implement the approach in each operator smaller,
+since we could conceivably wrap the existing static values with the
+function call.
+
+Users will specify the `controlPlaneCPULimit` when building a cluster
+by passing the value to the installer in `install-config.yaml`.
+
+### User Stories
+
+#### Radio Access Network (RAN) Use Case
+
+In the context of telcocommunications service providers' 5G Radio Access
+Networks, it is increasingly common to see "cloud native" implementations
+of the 5G Distributed Unit (DU) component. Due to latency constraints,
+this DU component needs to be deployed very close to the radio antenna for
+which it is responsible. In practice, this can mean running this
+component on anything from a single server at the base of a remote cell
+tower or in a datacenter-like environment serving several base stations.
+
+A hypothetical DU example is an unusually resource-intensive workload,
+requiring 20 dedicated cores, 24 GiB of RAM consumed as huge pages,
+multiple SR-IOV NICs carrying several Gbps of traffic each, and
+specialized accelerator devices. The node hosting this workload must
+run a realtime kernel, be carefully tuned to ensure low-latency
+requirements can be met, and be configured to support features like
+Precision Timing Protocol (PTP).
+
+The most constrained resource for RAN deployments is CPU, so for now
+the focus of this enhancement is scaling CPU requests.  Kubernetes
+resource requests for CPU time are measured in fractional "[Kubernetes
+CPUs](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu)". For
+bare metal, 1 Kubernetes CPU is equivalent to 1 hyperthread.
+
+Due to the resource-intensive workload, the overhead for platform
+components such as the OpenShift platform and add-on operators is
+severely restricted when compared to other uses. For example, one
+customer has allocated 4 hyperthreads for all components other than
+their own workloads. OpenShift currently requires 7.
+
+### Implementation Details/Notes/Constraints
+
+1. Add the new field to the `Infrastructure` API.
+2. Expose the `controlPlaneCPULimit` setting through the installer via
+   `install-config.yaml`.
+3. Update operators to respond to the API, as described above.
+
+### Risks and Mitigations
```

Comment:
@browsell as we discussed earlier today, I've added info about starting by not updating all operators at once. Rather than listing it as a complete separate 3rd option, I have it here as mitigation of the risk of the amount of effort of implementing the same approach across all operators. Please let me know what you think.

Reply:
Thanks. lgtm